### PR TITLE
filter private or archived channels from random channel

### LINKED ISSUE #393: Featured channels should not feature archived channels
**Describe the bug**
Featured channels should not feature archived channels. Because they were featured.

**To Reproduce**
1. Wait a bit
2. See error

**Expected behavior**
To not feature the archived channel.

**Screenshots**

### PR DESCRIPTION
**Pull request by angeleneleow**
# Pull Request

## Description

filter private/ archived channels from random channel

## Testing

no

## Ticket(s)

Closes #393 

(Create a copy of that line for each Github Issue affected,
and replace "Affects" with "Closes" if merging this will close the relevant ticket.)

### CODE REVIEW

**Comment by angeleneleow** [comment]
please i need hacktoberfest shirt :(

**Comment by bobheadxi** [comment]
@cheukyin699 this PR comes from a fork, it looks like build wont run because the workflow only runs on `push`, and not on `pull_request`:

https://github.com/ubclaunchpad/rocket2/blob/master/.github/workflows/pythonpackage.yml#L3

File: tests/app/scheduler/modules/random_channel_test.py [hunk 1]
```diff
@@ -12,13 +12,13 @@ def test_doing_it(bot, config, app):
     config.slack_notification_channel = ''
     config.slack_announcement_channel = ''
     bot.get_channels.return_value = [{'id': '123', 'name': 'general'},
-                                     {'id': '321', 'name': 'random'}]
+                                     {'id': '321', 'name': 'random', 'is_archived': True},
+                                     {'id': '456', 'name': 'bobheadxi', 'is_private': True}]
```
Comment (cheukyin699) [inline tests/app/scheduler/modules/random_channel_test.py]: These 2 lines are too long. We are having a bit of trouble with the CI at the moment, so bare with us.

You can run the tests that our CI runs with `scripts/build_check.sh` in your terminal.

**Comment by cheukyin699** [comment]
> @cheukyin699 this PR comes from a fork, it looks like build wont run because the workflow only runs on `push`, and not on `pull_request`:
> 
> [/.github/workflows/pythonpackage.yml@`master`#L3](https://github.com/ubclaunchpad/rocket2/blob/master/.github/workflows/pythonpackage.yml#L3) [](/ubclaunchpad/rocket2/blob/HEAD@{2019-10-31T04:35:36Z}/.github/workflows/pythonpackage.yml#L3)

On it! See #402

**Comment by cheukyin699** [comment]
Hey, will you continue working on this? I'll be closing this in 3 days if there is no activity.

**Comment by cheukyin699** [comment]
Closing because stale.

**Pull request closed without merging**